.github/workflows/python-dev.yml

```diff
@@ -75,8 +75,7 @@ jobs:
         python -m pip install --upgrade pip setuptools wheel
         python -m pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
         python -m pip install git+https://github.com/nedbat/coveragepy.git
-        python -m pip install versioneer[toml]
-        python -m pip install python-dateutil pytz cython hypothesis==6.52.1 pytest>=6.2.5 pytest-xdist pytest-cov pytest-asyncio>=0.17
+        python -m pip install -r ci/deps/requirements-minimal.txt pytest-cov pytest-xdist
```

Comment:
On second thought, are you sure pytest-xdist is not required? I remember having to install it once for some test(it was using one of the pytest-xdist fixtures).

Reply:
Yeah the S3 tests use the `worker_id` pytest-xdist fixture. All of them should be marked with `single_cpu` so I guess in this job those should be skipped. Might make sense to include pytest-xdist just in case anything changes